source/sliding_puzzle.py

```diff
@@ -124,7 +139,88 @@ def _join(self, row: List[PuzzlePiece]) -> List[str]:
     def draw(self) -> str:
         """Draw the puzzle in its current state."""
         output = []
-        for row in self.pieces:
+        for row in self.rows:
             output.extend(self._join(row))
 
         return '\n'.join(output)
+
+    def shuffle(self) -> None:
+        """Shuffle the puzzle by randomizing the order of the pieces"""
+        for row in self.rows:
+            random.shuffle(row)
+        puzzle_heigth, puzzle_width = len(self.rows), len(self.rows[0])
+        # empty one of the rows
+        x = random.choice(range(puzzle_heigth))
+        y = random.choice(range(puzzle_width))
+        self.rows[x][y].clear()
```

Comment:
You should be able to simplify this whole last part as: `choice(choice(self.rows)).clear()`

Reply:
yep, it looks better